packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -116,9 +116,6 @@ export interface VM<N = HostNode, E = HostElement> {
     /** Rendering operations associated with the VM */
     readonly renderMode: RenderMode;
     shadowMode: ShadowMode;
-    /** Transitive support for native Shadow DOM. A component in native mode
-     * transitively opts all of its descendants into native. */
-    readonly nearestShadowMode: ShadowMode | null;
```

Comment:
The problem with `nearestShadowMode` is that it would actually skip checking the immediate parent's `shadowMode`. Example:

- grandparent: `shadowSupportMode: reset`
- parent: `shadowSupportMode: any`
- child: `shadowSupportMode: reset`

In this case, the child would compute its `shadowMode` in `computeShadowMode` based on its parent's `nearestShadowMode` (see `owner?.nearestShadowMode ?? ShadowMode.Synthetic` below). In the above case, the parent's `nearestShadowMode` would be `Synthetic` rather than `Native`, because the grandparent is synthetic.

I couldn't see a reason to keep `nearestShadowMode`, so I removed it entirely and the tests all seem to be passing. It seems simplest to just get the shadow mode directly from the owner.